src/Umbraco.Infrastructure/Security/MembersUserStore.cs

```diff
@@ -0,0 +1,703 @@
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            if (login == null)
+            {
+                throw new ArgumentNullException(nameof(login));
+            }
+
+            ICollection<IIdentityUserLogin> logins = user.Logins;
+            var instance = new IdentityUserLogin(login.LoginProvider, login.ProviderKey, user.Id.ToString());
+            IdentityUserLogin userLogin = instance;
+            logins.Add(userLogin);
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public override Task RemoveLoginAsync(MembersIdentityUser user, string loginProvider, string providerKey, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            IIdentityUserLogin userLogin = user.Logins.SingleOrDefault(l => l.LoginProvider == loginProvider && l.ProviderKey == providerKey);
+            if (userLogin != null)
+            {
+                user.Logins.Remove(userLogin);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public override Task<IList<UserLoginInfo>> GetLoginsAsync(MembersIdentityUser user, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            return Task.FromResult((IList<UserLoginInfo>)user.Logins.Select(l => new UserLoginInfo(l.LoginProvider, l.ProviderKey, l.LoginProvider)).ToList());
+        }
+
+
+        /// <inheritdoc />
+        protected override async Task<IdentityUserLogin<string>> FindUserLoginAsync(string userId, string loginProvider, string providerKey, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            MembersIdentityUser user = await FindUserAsync(userId, cancellationToken);
+            if (user == null)
+            {
+                return null;
+            }
+
+            IList<UserLoginInfo> logins = await GetLoginsAsync(user, cancellationToken);
+            UserLoginInfo found = logins.FirstOrDefault(x => x.ProviderKey == providerKey && x.LoginProvider == loginProvider);
+            if (found == null)
+            {
+                return null;
+            }
+
+            return new IdentityUserLogin<string>
+            {
+                LoginProvider = found.LoginProvider,
+                ProviderKey = found.ProviderKey,
+                ProviderDisplayName = found.ProviderDisplayName, // TODO: We don't store this value so it will be null
```

Comment:
This is shared from the back office implementation and  is part of Microsoft IdentityUserLogin that _"Gets or sets the friendly name used in a UI for this login."_ but it isn't apparently used. It isn't set in your [UmbracoIdentity package](https://github.com/Shazwazza/UmbracoIdentity/blob/c12d07e1d3bf1aa2a7bdaba8184b1eacec9c0bce/src/UmbracoIdentity/UmbracoMembersUserStore.cs) either, so maybe unneeded? It would be a string more things like "Google", "local" etc. 

However, some of these external login _FindUserLoginAsync_ and similar methods need reviewing; they may be unwanted or could be combined during the code combining refactor.

Adding and removing external user logins etc hasn't yet been implemented or tested. I'm only implementing this and similar methods for members user store at this stage because this class inherits from identity _UserStoreBase_ and I thought I'd make a start on adding the body of some of the methods, following the `BackOfficeUserStore` approach.
 
It might be safer to throw a not implemented exception before they're implemented? 

I'm not sure the final plan for external logins for members yet so I'm happy to amend as needed.

Reply:
Cool, we'll revisit this when we do external logins